source/util/timer.cpp

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2018 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "util/timer.h"
+
+#include <sys/resource.h>
+#include <sys/time.h>
+#include <iomanip>
+#include <iostream>
+#include <map>
+#include <string>
+
+namespace spvutils {
+
+#if defined(SPIRV_TIMER_ENABLED)
+
+// A map data structure to search a CumulativeTimer object by its name. The key
+// of the map is the name and the value is the CumulativeTimer object.
+static std::map<const char*, CumulativeTimer*> CumulativeTimerMap;
```

Comment:
This is not allowed by the style guide.
https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables

But parts of this project don't follow the style guide...

Reply:
Hmm .. I cannot understand how to change this into the correct way. If I replace it with an array of pairs, it is not clear whether the length of array must be fixed or not. If I want to not fixed one, it looks like an implementation of the data structure from scratch.

Reply:
Thanks! I followed the first way (i.e., drop the map structure and functions related to it).